static/app/components/issueSyncListElement.tsx

```diff
@@ -59,73 +63,58 @@ class IssueSyncListElement extends Component<Props> {
       case 'jira_server':
         return 'Jira Server';
       default:
-        return capitalize(type);
-    }
-  }
-
-  getLink(): React.ReactElement {
-    return (
-      <IntegrationLink
-        href={this.props.externalIssueLink || undefined}
-        onClick={!this.isLinked() ? this.props.onOpen : undefined}
-        disabled={this.props.disabled}
-      >
-        {this.getText()}
-      </IntegrationLink>
-    );
-  }
-
-  getText(): React.ReactNode {
-    if (this.props.children) {
-      return this.props.children;
-    }
-    if (this.props.externalIssueDisplayName) {
-      return this.props.externalIssueDisplayName;
-    }
-    if (this.props.externalIssueKey) {
-      return this.props.externalIssueKey;
+        return capitalize(integrationType);
     }
-
-    return `${this.getPrettyName()} Issue`;
-  }
-
-  render() {
-    return (
-      <IssueSyncListElementContainer>
-        <ClassNames>
-          {({css}) => (
-            <StyledHovercard
-              containerClassName={css`
-                display: flex;
-                align-items: center;
-                min-width: 0; /* flex-box overflow workaround */
-
-                svg {
-                  flex-shrink: 0;
-                }
-              `}
-              header={this.props.hoverCardHeader}
-              body={this.props.hoverCardBody}
-              bodyClassName="issue-list-body"
-              forceVisible={this.props.showHoverCard}
-            >
-              {this.getIcon()}
-              {this.getLink()}
-            </StyledHovercard>
-          )}
-        </ClassNames>
-        {(this.props.onClose || this.props.onOpen) && (
-          <StyledIcon
-            role="button"
-            aria-label={this.isLinked() ? t('Close') : t('Add')}
-            onClick={this.handleIconClick}
+  }, [integrationType]);
+
+  const text =
+    children || externalIssueDisplayName || externalIssueKey || `${prettyName} Issue`;
+
+  const link = (
```

Comment:
is this being used somewhere else too? I would just move this to line 103

Reply:
I just assume it was extracted to make the react tree a bit easier to read